dts/arm/renesas/ra/ra-cm4-common.dtsi

```diff
@@ -26,6 +26,109 @@
 	};
 
 	soc {
+		cgc: cgc@4001e000 {
+			compatible = "renesas,ra-clock-generation-circuit";
+			reg = <0x4001e000 0x40 0x40047000 0x10>;
+			reg-names = "system", "mstp";
+			#clock-cells = <1>;
+
+			mosc: mosc {
+				compatible = "fixed-clock";
+				clock-frequency = <1000000>;
+				status = "disabled";
+				#clock-cells = <0>;
+			};
+
+			sosc: sosc {
+				compatible = "fixed-clock";
+				clock-frequency = <32768>;
+				status = "disabled";
+				#clock-cells = <0>;
+			};
+
+			hoco: hoco {
+				compatible = "fixed-clock";
+				clock-frequency = <24000000>;
+				status = "disabled";
+				#clock-cells = <0>;
+			};
+
+			moco: moco {
+				compatible = "fixed-clock";
+				clock-frequency = <8000000>;
+				status = "disabled";
+				#clock-cells = <0>;
+			};
+
+			loco: loco {
+				compatible = "fixed-clock";
+				clock-frequency = <32768>;
+				status = "disabled";
+				#clock-cells = <0>;
+			};
+
+			subclk: subclk {
+				compatible = "fixed-clock";
+				clock-frequency = <32768>;
+				status = "disabled";
+				#clock-cells = <0>;
+			};
+
+			pll: pll {
+				compatible = "fixed-factor-clock";
+				status = "disabled";
+				#clock-cells = <0>;
+			};
+
+			sysclk: sysclk {
+				compatible = "fixed-factor-clock";
+				status = "disabled";
+				#clock-cells = <0>;
+			};
+
+			iclk: iclk {
+				compatible = "fixed-factor-clock";
+				status = "disabled";
+				clocks = <&sysclk>;
+				#clock-cells = <0>;
+			};
+
+			pclka: pclka {
+				compatible = "fixed-factor-clock";
+				status = "disabled";
+				clocks = <&sysclk>;
+				#clock-cells = <0>;
+			};
+
+			pclkb: pclkb {
+				compatible = "fixed-factor-clock";
+				status = "disabled";
+				clocks = <&sysclk>;
+				#clock-cells = <0>;
+			};
+
+			pclkc: pclkc {
+				compatible = "fixed-factor-clock";
+				status = "disabled";
+				clocks = <&sysclk>;
+				#clock-cells = <0>;
+			};
+
+			pclkd: pclkd {
+				compatible = "fixed-factor-clock";
+				status = "disabled";
+				clocks = <&sysclk>;
+				#clock-cells = <0>;
+			};
+
+			fclk: fclk {
+				compatible = "fixed-factor-clock";
+				status = "disabled";
+				clocks = <&sysclk>;
+				#clock-cells = <0>;
+			};
```

Comment:
Wow, this block is super verbose, especially on the board file, you'll end up with a ton of nodes. How about dropping the child noes and just using properties for these? That's what the stm32 rcc block does https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/arduino_giga_r1/arduino_giga_r1.dtsi#L33, you could make them optional and use that instead of the disabled nodes, I think the configuration would become a lot easier to understand.

Reply:
The clock source was moved to the `clocks { ...}` section, and the division setting was replaced with the clock controller property.
It has cleared up a lot.